Index: src/main/java/com/example/demoweb/service/PostService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/example/demoweb/service/PostService.java	(revision 3c07bad6a8378982e69937b867868b6cd0d9aa37)
+++ src/main/java/com/example/demoweb/service/PostService.java	(date 1568261587000)
@@ -1,6 +1,8 @@
 package com.example.demoweb.service;
 
 import com.example.demoweb.model.Post;
+import com.example.demoweb.repository.PostRepository;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
@@ -10,22 +12,16 @@
 @Service
 public class PostService {
 
-    private List<Post> posts;
+    @Autowired
+    PostRepository postRepository;
 
-    {
-        posts = new ArrayList<>();
-        posts.add(new Post(0L,"пост 1", new Date()));
-        posts.add(new Post(1L,"пост 2", new Date()));
-        posts.add(new Post(2L,"пост 3", new Date()));
-    }
-
-    public List<Post> listAllPosts() {
-        return posts;
+    public Iterable<Post> listAllPosts() {
+        return postRepository.findAll();
     }
 
     public void create(String text) {
-        long newId = posts.size();
-        posts.add(new Post(newId, text, new Date()));
+        Post post = new Post(null, text, new Date());
+        postRepository.save(post);
     }
 
 }
Index: src/main/resources/application.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/resources/application.properties	(revision 3c07bad6a8378982e69937b867868b6cd0d9aa37)
+++ src/main/resources/application.properties	(date 1568260339000)
@@ -1,1 +1,8 @@
 server.port = 8000
+
+spring.datasource.url=jdbc:h2:file:./database
+spring.datasource.driverClassName=org.h2.Driver
+spring.datasource.username=root
+spring.datasource.password=root
+spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
+spring.jpa.hibernate.ddl-auto=update
\ No newline at end of file
Index: src/main/java/com/example/demoweb/model/Post.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/example/demoweb/model/Post.java	(revision 3c07bad6a8378982e69937b867868b6cd0d9aa37)
+++ src/main/java/com/example/demoweb/model/Post.java	(date 1568261500000)
@@ -1,14 +1,24 @@
 package com.example.demoweb.model;
 
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import java.io.Serializable;
 import java.util.Date;
 
-public class Post {
+@Entity
+public class Post implements Serializable {
 
+    @Id
+    @GeneratedValue
     private Long id;
     private String text;
     private int likes;
     private Date creationDate;
 
+    public Post() {
+    }
+
     public Post(Long id, String text, Date creationDate) {
         this.id = id;
         this.text = text;
Index: pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- pom.xml	(revision 3c07bad6a8378982e69937b867868b6cd0d9aa37)
+++ pom.xml	(date 1568261366000)
@@ -34,6 +34,16 @@
 			<scope>test</scope>
 		</dependency>
 
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+            <scope>runtime</scope>
+        </dependency>
+
 	</dependencies>
 
 	<build>
Index: src/main/java/com/example/demoweb/DemoWebApplication.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/example/demoweb/DemoWebApplication.java	(revision 3c07bad6a8378982e69937b867868b6cd0d9aa37)
+++ src/main/java/com/example/demoweb/DemoWebApplication.java	(date 1568218370000)
@@ -2,12 +2,23 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.ConfigurableApplicationContext;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
 
 @SpringBootApplication
 public class DemoWebApplication {
 
-	public static void main(String[] args) {
-		SpringApplication.run(DemoWebApplication.class, args);
+	public static void main(String[] args) throws UnknownHostException {
+		ConfigurableApplicationContext context = SpringApplication.run(DemoWebApplication.class, args);
+
+		String port = context.getEnvironment().getProperty("server.port");
+
+		InetAddress ip = InetAddress.getLocalHost();
+		String hostname = ip.getHostName();
+
+		System.out.println("http://localhost:" + port + "/");
 	}
 
 }
Index: src/main/java/com/example/demoweb/repository/PostRepository.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/example/demoweb/repository/PostRepository.java	(date 1568260629000)
+++ src/main/java/com/example/demoweb/repository/PostRepository.java	(date 1568260629000)
@@ -0,0 +1,7 @@
+package com.example.demoweb.repository;
+
+import com.example.demoweb.model.Post;
+import org.springframework.data.repository.CrudRepository;
+
+public interface PostRepository extends CrudRepository<Post, Long> {
+}
\ No newline at end of file
Index: src/main/java/com/example/demoweb/service/LikesService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/example/demoweb/service/LikesService.java	(revision 3c07bad6a8378982e69937b867868b6cd0d9aa37)
+++ src/main/java/com/example/demoweb/service/LikesService.java	(date 1568261469000)
@@ -1,6 +1,7 @@
 package com.example.demoweb.service;
 
 import com.example.demoweb.model.Post;
+import com.example.demoweb.repository.PostRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
@@ -8,11 +9,12 @@
 public class LikesService {
 
     @Autowired
-    PostService postService;
+    PostRepository postRepository;
 
     public int like(Long postId) {
-        Post post = postService.listAllPosts().get(postId.intValue());
+        Post post = postRepository.findById(postId).get();
         post.setLikes(post.getLikes() + 1);
+        postRepository.save(post);
         return post.getLikes();
     }
 
